engine/src/main/java/org/hibernate/search/engine/search/dsl/predicate/impl/RangePredicateFieldSetContextImpl.java

```diff
@@ -92,8 +92,8 @@ public void contributePredicateBuilders(Consumer<B> collector) {
 		}
 	}
 
-	static class CommonState<B> extends AbstractBooleanMultiFieldPredicateCommonState<B, RangePredicateFieldSetContextImpl<B>>
-			implements RangePredicateTerminalContext {
+	static class CommonState<B> extends AbstractBooleanMultiFieldPredicateCommonState<CommonState<B>, B, RangePredicateFieldSetContextImpl<B>>
+			implements RangePredicateTerminalContext, RangePredicateLimitTerminalContext {
```

Comment:
Maybe we could remove the first interface after implements here.

Reply:
Done